infra/stages/staging/main.tf

```diff
@@ -18,18 +18,70 @@ provider "aws" {
   region  = var.aws_region
 }
 
+module "concrexit_network" {
+  source = "../../modules/concrexit_network"
+  stage  = "staging"
+  tags = {
+    "Category"    = "concrexit",
+    "Owner"       = "technicie",
+    "Environment" = "staging",
+    "Terraform"   = true
+  }
+  customer = var.customer
+}
+
 module "concrexit" {
-  source               = "../../modules/concrexit"
-  stage                = "staging"
-  tags                 = {
+  source = "../../modules/concrexit"
+  depends_on = [
+    aws_route53_record.www,
+    aws_route53_record.wildcard
+  ]
+  stage = "staging"
+  tags = {
     "Category"    = "concrexit",
     "Owner"       = "technicie",
     "Environment" = "staging",
     "Terraform"   = true
   }
-  customer             = var.customer
-  webhostname          = "staging"
-  domain               = var.domain_name
-  ssh_private_key      = var.ssh_private_key
-  ssh_public_key       = var.ssh_public_key
+  customer         = var.customer
+  webhostname      = "staging"
+  domain           = var.domain_name
+  ssh_private_key  = var.ssh_private_key
+  ssh_public_key   = var.ssh_public_key
+  aws_interface_id = module.concrexit_network.aws_interface_id
+  # We supply the ipv4 address to this module and get ipv6 address back
+  # This weird construction is because only when creating the instance is the
+  # ipv6 address known, but the elastic IP is created beforehand.
+  public_ipv4      = module.concrexit_network.public_ipv4
+}
+
+data "aws_route53_zone" "primary" {
+  name = var.domain_name
+}
+
+resource "aws_route53_record" "www" {
```

Comment:
Feels really weird to have these here, are they not part of concrexit?

Reply:
They will be different for staging/production/review so I could enable them conditionally in the concrexit module or create a new module but this also seemed fine

Reply:
Feels like a lot of duplication, since the zone_id, name and ipv4 can just be provided to the module itself.

Reply:
If anything, this could probably be a `concrexit-domain` module or something. That way I'd be happy.

Reply:
This is also needed because the letsencrypt stuff in the nixos config will only work if the A record exists, so those are given as dependencies to the concrexit module